### PR TITLE
Complete PostgreSQL `CREATE TYPE` Support

### PR DESCRIPTION
**Pull request by LucaCappelletti94**
Adds full parsing for all PostgreSQL `CREATE TYPE` statement variants. Previously only composite and enum types were supported.

## What's New

**Range Types** - Parse range type definitions with 6 configuration options

```sql
CREATE TYPE int4range AS RANGE (SUBTYPE = int4, CANONICAL = fn);
```

**Base Types** - Parse low-level type definitions with 19 options (INPUT/OUTPUT functions, ALIGNMENT, STORAGE, etc.)

```sql
CREATE TYPE complex (INPUT = in_fn, OUTPUT = out_fn, INTERNALLENGTH = 16);
```

**Simple Declarations** - Parse type shells without representation

```sql
CREATE TYPE typename;
```

The AST extends `UserDefinedTypeRepresentation` with three new variants (`Range`, `SqlDefinition`, `None`) and adds five supporting enums (`UserDefinedTypeRangeOption`, `UserDefinedTypeSqlDefinitionOption`, `Alignment`, `UserDefinedTypeStorage`, `UserDefinedTypeInternalLength`) with comprehensive rustdoc. The parser refactors `parse_create_type()` into modular functions (`parse_create_type_composite()`, `parse_create_type_range()`, and base type option parsers) that handle all PostgreSQL type configuration syntax. Tests provide comprehensive coverage with explicit AST verification for all variants and edge cases. Added 17 PostgreSQL-specific keywords, necessary to parse the keywords used in the type options (ALIGNMENT, CANONICAL, INTERNALLENGTH, PASSEDBYVALUE, RECEIVE, SEND, STORAGE, SUBTYPE, TYPMOD_IN, etc.).

Closes issue #2092

### CODE REVIEW

File: src/ast/ddl.rs [hunk 1]
```diff
@@ -2002,22 +2002,45 @@ impl fmt::Display for DropBehavior {
...
+    /// Range type: `CREATE TYPE name AS RANGE (options)`
+    Range {
+        options: Vec<UserDefinedTypeRangeOption>,
+    },
+    /// Base type (SQL definition): `CREATE TYPE name (options)`
+    ///
+    /// Note the lack of `AS` keyword
+    SqlDefinition {
+        options: Vec<UserDefinedTypeSqlDefinitionOption>,
+    },
```
Comment (iffyio) [inline src/ast/ddl.rs:2047]: Could we include a link to the postgres docs for the new variants?

File: src/ast/ddl.rs [hunk 2]
```diff
@@ -2002,22 +2002,45 @@ impl fmt::Display for DropBehavior {
...
+    /// When the representation of the type is not specified.
+    /// This is used in `CREATE TYPE <name>;` statements.
+    None,
```
Comment (iffyio) [inline src/ast/ddl.rs]: Instead of having a special `None` variant, ideally the usage becomes an optional field, would that work here? i.e `representation: Option<UserDefinedTypeRepresentation>`

File: src/parser/mod.rs [hunk 3]
```diff
@@ -17331,20 +17331,60 @@ impl<'a> Parser<'a> {
...
+            self.expected("ENUM, RANGE, or '(' after AS", self.peek_token())
         }
+    }
 
-        let mut attributes = vec![];
-        if !self.consume_token(&Token::LParen) || self.consume_token(&Token::RParen) {
+    /// Parse remainder of `CREATE TYPE AS (attributes)` statement (composite type)
+    ///
+    /// See [PostgreSQL](https://www.postgresql.org/docs/current/sql-createtype.html)
+    pub fn parse_create_type_composite(
```
Comment (iffyio) [inline src/parser/mod.rs]:
```suggestion
    fn parse_create_type_composite(
```

File: src/parser/mod.rs [hunk 4]
```diff
@@ -17387,6 +17425,254 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse remainder of `CREATE TYPE AS RANGE` statement
+    ///
+    /// See [PostgreSQL](https://www.postgresql.org/docs/current/sql-createtype.html)
+    pub fn parse_create_type_range(&mut self, name: ObjectName) -> Result<Statement, ParserError> {
```
Comment (iffyio) [inline src/parser/mod.rs]:
```suggestion
    fn parse_create_type_range(&mut self, name: ObjectName) -> Result<Statement, ParserError> {
```

File: src/parser/mod.rs [hunk 5]
```diff
@@ -17387,6 +17425,254 @@ impl<'a> Parser<'a> {
...
+            name,
+            representation: UserDefinedTypeRepresentation::Range { options },
+        })
+    }
+
+    /// Parse a single range option for CREATE TYPE AS RANGE
+    fn parse_range_option(&mut self) -> Result<UserDefinedTypeRangeOption, ParserError> {
+        let keyword = self.parse_one_of_keywords(&[
+            Keyword::SUBTYPE,
+            Keyword::SUBTYPE_OPCLASS,
```
Comment (iffyio) [inline src/parser/mod.rs:17473]: can we add tests for `SUBTYPE_OPCLASS`,`SUBTYPE_DIFF` and any other variants that are missing coverage?

File: src/parser/mod.rs [hunk 6]
```diff
@@ -17387,6 +17425,254 @@ impl<'a> Parser<'a> {
...
+        let options = self.parse_comma_separated0(|p| p.parse_range_option(), Token::RParen)?;
+        self.expect_token(&Token::RParen)?;
+
+        Ok(Statement::CreateType {
+            name,
+            representation: UserDefinedTypeRepresentation::Range { options },
+        })
+    }
+
+    /// Parse a single range option for CREATE TYPE AS RANGE
```
Comment (iffyio) [inline src/parser/mod.rs]:
```suggestion
    /// Parse a single range option for a `CREATE TYPE AS RANGE` statement
```

**Review by iffyio — COMMENTED** [review]
Thanks @LucaCappelletti94! Changes look good to me overall, left some comments

File: src/ast/ddl.rs (hunk 1 above)
Comment (LucaCappelletti94) [reply src/ast/ddl.rs:2047]: I wanted to, but they do not have a linkable subsection so the link would be to the same `CREATE TYPE` page. The sub-headers do not have the anchor link unfortunately.

File: src/ast/ddl.rs (hunk 2 above)
Comment (LucaCappelletti94) [reply src/ast/ddl.rs]: Sure I can switch to that case - should we normalize other analogous enumerations which also have the `None` field, maybe in another PR?

File: src/parser/mod.rs (hunk 5 above)
Comment (LucaCappelletti94) [reply src/parser/mod.rs:17473]: Oke

File: src/ast/ddl.rs (hunk 2 above)
Comment (LucaCappelletti94) [reply src/ast/ddl.rs]: Done

File: src/parser/mod.rs (hunk 5 above)
Comment (LucaCappelletti94) [reply src/parser/mod.rs:17473]: Tentatively done, I think all are covered now.

File: src/ast/ddl.rs (hunk 1 above)
Comment (iffyio) [reply src/ast/ddl.rs:2047]: ah I think in that case linking to the postgres create type page for each variant would suffice, most important that its clear where each variant comes from especially if the overlying statement/enum is shared between multiple dialects. Basically similar to what the `Enum { ... }` variant above documents?

Comment (LucaCappelletti94) [reply src/ast/ddl.rs:2047]: Ok, doing that.

Comment (LucaCappelletti94) [reply src/ast/ddl.rs:2047]: Added

**Review by iffyio — APPROVED** [review]
LGTM! Thanks @LucaCappelletti94!
